python-avd/pyavd/_eos_cli_config_gen/j2templates/eos/ethernet-interfaces.j2

```diff
@@ -990,6 +990,122 @@ interface {{ ethernet_interface.name }}
+{%             endif %}
+   {{ tap_mac_address_cli }}
+{%         endif %}
+{# The position of "mpls_pop_all" does not match the EOS CLI to implement this "elif" with "vxlan_strip" #}
+{%         if ethernet_interface.switchport.tap.mpls_pop_all is arista.avd.defined(true) %}
+   switchport tap mpls pop all
+{%         elif ethernet_interface.switchport.tap.encapsulation.vxlan_strip is arista.avd.defined(true) %}
+   switchport tap encapsulation vxlan strip
+{%         endif %}
+{%         for protocol in ethernet_interface.switchport.tap.encapsulation.gre.protocols | arista.avd.natural_sort('protocol') %}
+{%             set tap_encapsulation_cli = "switchport tap encapsulation gre protocol " ~ protocol.protocol %}
+{%             if protocol.feature_header_length is arista.avd.defined %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ protocol.feature_header_length %}
+{%             endif %}
+{%             set tap_encapsulation_cli = tap_encapsulation_cli ~ " strip" %}
+{%             if protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " re-encapsulation ethernet" %}
+{%             endif %}
+   {{ tap_encapsulation_cli }}
+{%         endfor %}
+{%         if ethernet_interface.switchport.tap.encapsulation.gre.strip is arista.avd.defined(true) %}
+   switchport tap encapsulation gre strip
+{%         endif %}
+{%         for destination in ethernet_interface.switchport.tap.encapsulation.gre.destinations | arista.avd.natural_sort('destination') %}
+{%             set tap_encapsulation_cli = "switchport tap encapsulation gre destination " ~ destination.destination %}
+{%             if destination.source is arista.avd.defined %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " source " ~ destination.source %}
+{%             endif %}
+{%             for destination_protocol in destination.protocols | arista.avd.natural_sort('protocol') %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " protocol " ~ destination_protocol.protocol %}
+{%                 if destination_protocol.feature_header_length is arista.avd.defined %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " feature header length " ~ destination_protocol.feature_header_length %}
+{%                 endif %}
+{%                 set tap_encapsulation_cli = tap_encapsulation_cli ~ " strip" %}
+{%                 if destination_protocol.re_encapsulation_ethernet_header is arista.avd.defined(true) %}
+{%                     set tap_encapsulation_cli = tap_encapsulation_cli ~ " re-encapsulation ethernet" %}
+{%                 endif %}
+   {{ tap_encapsulation_cli }}
+{%             endfor %}
+{%             if destination.protocols is not arista.avd.defined %}
```

Comment:
even if protocol is defined we can have separate strip command for same destination.
```
switchport tap encapsulation gre destination 10.1.1.1 source 11.1.1.1 protocol 0xffff feature header length 11 strip re-encapsulation ethernet
switchport tap encapsulation gre destination 10.1.1.1 source 11.1.1.1 strip
```
I think you just need to add strip key for every destination, then just check `if destination.strip is arista.avd.defined`